### ISSUE TITLE
[Java] Add Unicode Bypass Validation query, test and help file

### ISSUE THREAD

**Issue by Sim4n6** [github/securitylab#752]
### Query PR

https://github.com/github/codeql/pull/12995

### Language

Java

### CVE(s) ID list

- [CVE-2019-10160](https://github.com/advisories/GHSA-379v-rm3f-c48g)

### CWE

CWE-176

### Report

Please refer to the report in this issue: https://github.com/github/securitylab/issues/749

### Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc).

- [X] Yes
- [ ] No

### Blog post link

https://sim4n6.beehiiv.com

**Comment by jorgectf** [comment]
👋 @Sim4n6

Thank you for your contribution!

The CVE linked does not relate to the contribution itself, as the code in question is not Java. Could you provide a CVE that matches the contribution fully?

**Comment by smowton** [comment]
@Sim4n6 given you are still working on the Java QL, just noting the outstanding CVE request above as well.

**Comment by Sim4n6** [comment]
No problem , I've got a hit on a codebase ... I 'm still working with the maintainer about the fix.

**Comment by Sim4n6** [comment]
I have identified one in this [codebase](https://github.com/SerNet/verinice/), and coordinated with the maintainers via email about the disclosure. The fix is up [[commit](https://github.com/SerNet/verinice/commit/c864de86374237c91c9dc5376cbbee0ae4b4ab4b) and they are thankful but no CVE requested or advisory issued.

**Comment by Sim4n6** [comment]
I found one for this query [GHSA-mgqj-hphf-9588](https://github.com/lockss/lockss-daemon/security/advisories/GHSA-mgqj-hphf-9588).

**Comment by jorgectf** [comment]
👋 @Sim4n6 

Thanks for your reply. According to the `All for One` rules, _your query must find at least one CVE_, but the advisory you linked above has not been requested for a CVE.

Keep us posted!

**Comment by jorgectf** [comment]
It seems `CVE-2023-42183` hasn't been published yet, and it is not linked to https://github.com/lockss/lockss-daemon/security/advisories/GHSA-mgqj-hphf-9588, I0ll have to wait until either thing happens.

**Comment by Sim4n6** [comment]
yeah, I am waiting for that for sure. But it is reserved I 've got a message from them.

**Comment by Sim4n6** [comment]
And it's on... The [CVE-2023-42183](https://nvd.nist.gov/vuln/detail/CVE-2023-42183) is published by MITRE CNA. Thanks

**Comment by Sim4n6** [comment]
@jorgectf please, is it possible to have this handled before the new year?

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Query review**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Final decision**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Pay**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by xcorail** [comment]
Created Hackerone report 2293359 for bounty 539411 : [752] [Java] Add Unicode Bypass Validation query, test and help file

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Closed**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_
